### DISCUSSION TITLE
Difference between vrf_id and present_in_vrf ?

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#15843] [Q&A]
What's the difference between these 2 fields when searching for IP Addresses?  In the UI, `vrf_id` is called "Assigned VRF".
They appear to be exactly the same, but I haven't made extensive use of VRFs in Netbox yet.

![image](https://github.com/netbox-community/netbox/assets/3991046/58af139e-e140-48f1-9ef8-f6b32e7760ae)

**Comment by candlerb** [comment]
The form source is [here](https://github.com/netbox-community/netbox/blob/v3.7.6/netbox/ipam/forms/filtersets.py#L193-L203) and the filter source is [here](https://github.com/netbox-community/netbox/blob/v3.7.6/netbox/ipam/filtersets.py#L280-L300), with helper function `filter_present_in_vrf()` [here](https://github.com/netbox-community/netbox/blob/v3.7.6/netbox/ipam/filtersets.py#L420-L427).

The difference is that `present_in_vrf[_id]` shows IP addresses whose assigned VRF is the selected VRF *or* where their assigned VRF's export target(s) matches one of the selected VRF's import targets:

```
        return queryset.filter(
            Q(vrf=vrf) |
            Q(vrf__export_targets__in=vrf.import_targets.all())
        )
```
